components/style/gecko/arc_types.rs

```diff
@@ -0,0 +1,57 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+//! This file lists all arc FFI types and defines corresponding addref
+//! and release functions.
+
+#![allow(non_snake_case, missing_docs)]
+
+use gecko_bindings::bindings::{RawServoStyleSheet, RawServoStyleRule, RawServoImportRule};
+use gecko_bindings::bindings::{ServoComputedValues, ServoCssRules};
+use gecko_bindings::structs::{RawServoAnimationValue, RawServoDeclarationBlock};
+use gecko_bindings::sugar::ownership::{HasArcFFI, HasFFI};
+use parking_lot::RwLock;
+use properties::{ComputedValues, PropertyDeclarationBlock};
+use properties::animated_properties::AnimationValue;
+use stylesheets::{CssRules, Stylesheet, StyleRule, ImportRule};
+
+macro_rules! impl_arc_ffi {
```

Comment:
Might want to add more docs and note the corresponding macro to use on the gecko side.

Also differentiate it from https://github.com/upsuper/servo/blob/2fac64a3f610402c4f09e6e59e5bcb834acc03b4/components/style/gecko_bindings/sugar/refptr.rs#L255 (which goes in the opposite direction)